authors/apps/articles/views.py

```diff
@@ -365,3 +366,77 @@ def get(self, request, *args, **kwargs):
             },
                 status=status.HTTP_404_NOT_FOUND)
         return Response(serializer.data, status.HTTP_200_OK)
+
+
+class ReportArticleView(generics.ListCreateAPIView):
+    """
+    View for user to report article
+    """
+    permission_classes = (IsAuthenticated,)
+    serializer_class = ReportSerializer
+    renderer_classes = (ReportJSONRenderer,)
+
+    def get_object(self, *args, **kwargs):
+        slug = self.kwargs.get('slug')
+        try:
+            article = Article.objects.get(slug=slug)
+            return article
+
+        except Article.DoesNotExist:
+            raise NotFound(
+                {'error': 'Article does not exist'})
+
+    def get_queryset(self, request):
+        queryset = ReportArticle.objects.all()
+        article = self.get_object()
+        reported_article = queryset.filter(
+            article=article, reported_by=self.request.user).first()
+
+        return reported_article
```

Comment:
@VivianDoreen I think it would be better to let this method return a list of all the articles reported by a user. So that you don't need to override the get() method.

```
query_set = ReportArticle.objects.all()

def get_queryset(self, request):
    user = request.user
    return self.get_queryset.filter(reported_by=user)
```

Then when doing the post, to check whether an article has already been reported you could do

```
slug = self.kwargs.get('slug')
isReported = self.get_queryset().filter(slug=slug)
if isReported:
    ...
```